addons/event_sale/static/src/js/sale_product_field.js

```diff
@@ -59,3 +62,30 @@ patch(SaleOrderLineProductField.prototype, 'event_sale', {
         );
     },
 });
+
+export class SaleOrderLineProductQuantityField extends FloatField {
+    setup() {
+        super.setup();
+        this.action = useService("action");
+        this.orm = useService("orm");
+        this.uiService = useService("ui");
+        this.state = useState({ value: this.props.value });
+        useEffect(() => {
+            if ((this.state.value !== this.props.value || this.props.record.isNew) && this.props.record.data.state === 'sale' && this.props.record.data.event_id) {
+                this.state.value = this.props.value;
+                this.update();
+            }
+        });
+    }
+    async update() {
+        this.uiService.block();
+        await this.props.record.model.root.save({ stayInEdition: true });
+        const orderLineIds = this.props.record.model.root.data.order_line.resIds;
```

Comment:
afaik, `this.env.model` would point to the same model so maybe we can save a couple references ^^